client/www/pages/status.tsx

```diff
@@ -0,0 +1,399 @@
+import {
+  LandingContainer,
+  LandingFooter,
+  MainNav,
+} from '@/components/marketingUi';
+import * as og from '@/lib/og';
+import { useState, useEffect } from 'react';
+
+function getUptimeColor(percentage: number) {
+  if (percentage < 0 || percentage > 100) {
+    return '#e5e7eb';
+  }
+  if (percentage >= 100) {
+    return '#22c55e';
+  } else if (percentage >= 99.9) {
+    return '#86efac';
+  } else if (percentage >= 99) {
+    return '#fbbf24';
+  } else {
+    return '#fb923c';
+  }
+}
+
+function StatusPage() {
+  const [small, setSmall] = useState(false);
+  const [uptimeData, setUptimeData] = useState<any>(null);
+  const [loading, setLoading] = useState(true);
+  const [lastUpdated, setLastUpdated] = useState(new Date());
+  const [nextUpdate, setNextUpdate] = useState(60);
+
+  const processUptimeData = (rawData: any) => {
+    const monitors =
+      rawData.monitors?.map((monitor: any) => {
+        const customRatios = monitor.custom_uptime_ratio?.split('-') || [];
+        const customRanges = monitor.custom_uptime_ranges?.split('-') || [];
+
+        const dailyUptime = customRanges.map((range: string) => {
+          const uptime = parseFloat(range);
+          return isNaN(uptime) ? 100 : uptime;
+        });
+
+        while (dailyUptime.length < 90) {
+          dailyUptime.push(100);
+        }
+
+        return {
+          id: monitor.id,
+          friendly_name: monitor.friendly_name,
+          url: monitor.url,
+          status: monitor.status,
+          uptime_ratio: {
+            '24h': parseFloat(customRatios[0]) || 100,
+            '7d': parseFloat(customRatios[1]) || 100,
+            '30d': parseFloat(customRatios[2]) || 100,
+            '90d': parseFloat(customRatios[3]) || 100,
+            all_time: parseFloat(monitor.all_time_uptime_ratio) || 100,
+          },
+          daily_uptime: dailyUptime,
+          average_response_time: monitor.average_response_time,
+          logs: monitor.logs?.slice(0, 10) || [],
+        };
+      }) || [];
+
+    const overallUptime = {
+      '24h':
+        monitors.reduce(
+          (acc: number, m: any) => acc + m.uptime_ratio['24h'],
+          0,
+        ) / (monitors.length || 1),
+      '7d':
+        monitors.reduce(
+          (acc: number, m: any) => acc + m.uptime_ratio['7d'],
+          0,
+        ) / (monitors.length || 1),
+      '30d':
+        monitors.reduce(
+          (acc: number, m: any) => acc + m.uptime_ratio['30d'],
+          0,
+        ) / (monitors.length || 1),
+      '90d':
+        monitors.reduce(
+          (acc: number, m: any) => acc + m.uptime_ratio['90d'],
+          0,
+        ) / (monitors.length || 1),
+    };
+
+    return {
+      monitors,
+      overall_uptime: overallUptime,
+      last_updated: new Date().toISOString(),
+    };
+  };
+
+  const fetchUptimeData = async () => {
+    try {
+      const response = await fetch('/api/uptime');
+      const rawData = await response.json();
```

Comment:
nit: I would coerce this `rawData` into an UptimeRobotResponse or something like this. 

This way it will be easier for the next developer to know what fields were available.